src/lxc/log.c

```diff
@@ -562,42 +539,61 @@ static int _lxc_log_set_file(const char *name, const char *lxcpath, int create_d
 		ERROR("Could not build log path");
 		return -1;
 	}
+
 	ret = __lxc_log_set_file(logfile, create_dirs);
 	free(logfile);
 	return ret;
 }
 
-extern int lxc_log_syslog(int facility)
+static int lxc_log_enable_logfile(struct lxc_log *log)
 {
-	struct lxc_log_appender *appender;
+	int ret;
 
-	openlog(log_prefix, LOG_PID, facility);
-	if (!lxc_log_category_lxc.appender) {
-		lxc_log_category_lxc.appender = &log_appender_syslog;
-		return 0;
-	}
+	if (!log)
+		return -1;
 
-	appender = lxc_log_category_lxc.appender;
-	/* Check if syslog was already added, to avoid creating a loop */
-	while (appender) {
-		if (appender == &log_appender_syslog) {
-			/* not an error: openlog re-opened the connection */
+	if (log->file) {
+		if (strncmp(log->file, "none", strlen(log->file)) == 0)
```

Comment:
The `strlen()` is unnecessary overhead. `strcmp()` is safe to use.

Reply:
@brauner "lxc-unshare -o noneasdfasdfasdf" is also working. Is it allowed?

Reply:
Yes, a logfile named `noneasdfasdfasdf` should've been created.